features/admin.sms-templates.v1/components/sms-customization-form.tsx

```diff
@@ -0,0 +1,199 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import { Field, Form } from "@wso2is/form";
+import {
+    ContentLoader,
+    DocumentationLink,
+    Message,
+    useDocumentation
+} from "@wso2is/react-components";
+import React, {
+    FunctionComponent,
+    ReactElement,
+    useEffect,
+    useState
+} from "react";
+import { useTranslation } from "react-i18next";
+import { Grid } from "semantic-ui-react";
+import { SmsTemplate } from "../models/sms-templates";
+
+interface SmsCustomizationFormPropsInterface extends IdentifiableComponentInterface {
+    /**
+     * Is SMS templates list loading.
+     */
+    isSmsTemplatesListLoading: boolean;
+
+    /**
+     * Selected SMS template
+     */
+    selectedSmsTemplate: SmsTemplate;
+
+    /**
+     * Selected locale
+     */
+    selectedLocale: string;
+
+    /**
+     * Callback to be called when the template is changed.
+     * @param template - SMS template
+     */
+    onTemplateChanged: (updatedTemplateAttributes: Partial<SmsTemplate>) => void;
+
+    /**
+     * Callback to be called when the form is submitted.
+     */
+    onSubmit: () => void;
+
+    /**
+     * Callback to be called when the template delete requested
+     */
+    onDeleteRequested: () => void;
+
+    /**
+     * Is readonly.
+     */
+    readOnly?: boolean;
+}
+
+const FORM_ID: string = "sms-customization-content-form";
+
+/**
+ * SMS customization form.
+ *
+ * @param props - Props injected to the component.
+ *
+ * @returns SMS template customization form.
+ */
+export const SmsCustomizationForm: FunctionComponent<SmsCustomizationFormPropsInterface> = (
+    props: SmsCustomizationFormPropsInterface
+): ReactElement => {
+
+    const {
+        isSmsTemplatesListLoading,
+        selectedSmsTemplate,
+        onTemplateChanged,
+        onSubmit,
+        readOnly,
+        ["data-componentid"]: componentId
+    } = props;
+
+    const { t } = useTranslation();
+    const { getLink } = useDocumentation();
+
+    /**
+     * Following `key` state and the use of the useEffect are temporary
+     * fixes for the issue of the input not re-rendering when the props change.
+     * The ideal solution is to use Final Form directly without employing
+     * the Final Form wrapper component.
+     */
+    const [ key, setKey ] = useState<number>(0);
+
+    useEffect(() => {
+        setKey((key + 1) % 100);
+    }, [ selectedSmsTemplate ]);
+
+    return (
+        (isSmsTemplatesListLoading) ? (
+            <ContentLoader/>
+        ) : (
+            <>
+                <Form
+                    id={ FORM_ID }
+                    uncontrolledForm={ true }
+                    onSubmit={ onSubmit }
+                    data-componentid={ componentId }
+                >
```

Comment:
Better formatting suggestion.

```suggestion

    if (isSmsTemplatesListLoading) {
    	return <ContentLoader/>;
    }
    
    return (
        <>
                <Form
                    id={ FORM_ID }
                    uncontrolledForm={ true }
                    onSubmit={ onSubmit }
                    data-componentid={ componentId }
                >
```